src/rshim.c

```diff
@@ -2084,9 +2096,100 @@ rshim_backend_t *rshim_find_by_dev(void *dev)
   return NULL;
 }
 
-/* House-keeping timer. */
-static void rshim_timer_func(rshim_backend_t *bd)
+int rshim_set_drop_mode(rshim_backend_t *bd, int value)
+{
+  int old_value;
+  int rt = 0;
+
+  pthread_mutex_lock(&bd->mutex);
+  old_value = (int)bd->drop_mode;
+  value = !!value;
+  if (value == old_value) {
+    pthread_mutex_unlock(&bd->mutex);
+    return -EALREADY;
+  }
+
+  bd->drop_mode = 0;
+  if (bd->enable_device && bd->enable_device(bd, value ? false : true))
+    bd->drop_mode = 1;
+  else
+    bd->drop_mode = value;
+
+  if (bd->drop_mode)
+    bd->drop_pkt = 1;
+  else
+    rshim_fifo_sync(bd, true);
+  pthread_mutex_unlock(&bd->mutex);
+  /*
+   * Check if another endpoint driver has already attached to the
+   * same rshim device before enabling it.
+   */
+  if (!bd->drop_mode) {
+    rshim_lock();
+    pthread_mutex_lock(&bd->mutex);
+    if (rshim_access_check(bd)) {
+      RSHIM_WARN("rshim%d is not accessible\n", bd->index);
+      bd->drop_mode = 1;
+      rt = -EACCES;
+    }
+    pthread_mutex_unlock(&bd->mutex);
+    rshim_unlock();
+  }
+  return rt;
+}
+
+static int rshim_check_locked_mode(rshim_backend_t *bd)
+{
+    int rc;
+    uint64_t value;
+
+    rc = bd->read_rshim(bd, RSHIM_CHANNEL, bd->regs->scratchpad1, &value,
+                        RSHIM_REG_SIZE_8B);
+    if (rc < 0) {
+        RSHIM_ERR("RSHIM %d SCRATCHPAD1 register read error\n", bd->index);
+        return -EIO;
+    }
+
+    return (value == BF3_RSH_SECURE_NIC_MODE_MAGIC_NUM) ? 1 : 0;
+}
+
+static int rshim_update_locked_mode(rshim_backend_t *bd)
 {
+  int rt;
+  bool locked_mode;
+
+  rt = rshim_check_locked_mode(bd);
+  if (rt < 0) { /* Failed to check NIC locked mode status */
+    return -EIO;
+  }
+
+  locked_mode = !!rt;
+
```

Comment:
These two lines seems unnecessary since the rshim_check_locked_mode() returns <0, 0 or 1.